### PR TITLE
update node to v14

### PR DESCRIPTION
**Pull request by nl-ria**
## Why?

Tech Debt week

## What?

- Updated rel-engage to v8
- Updated node to v14 in both circleCI's docker image and in package.json
- Added the missing PR template
-  Dropped checking for compatibility of node v10 since its going [End of Life](https://nodejs.org/en/about/releases/) in April 2021
- linting changes after running make verify 

### Anything in particular you'd like to highlight to reviewers?

I learnt that Dubnium and Erbium are node version codenames ! This sounds like a major release?

### CODE REVIEW

**Comment by nl-ria** [comment]
Hmmm.. circleci build is still failing.. 
ideas?
```
make build
internal/modules/cjs/loader.js:807
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: The module '/home/circleci/project/lambda-logger/node_modules/iltorb/build/bindings/iltorb.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 83. This version of Node.js requires
NODE_MODULE_VERSION 64. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:807:18)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/circleci/project/lambda-logger/node_modules/iltorb/index.js:10:40)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
Makefile:27: recipe for target 'build' failed
make: *** [build] Error 1

Exited with code exit status 2
CircleCI received exit code 2
```
I have tried to bust circleCI cache by changing the cache key, but the error is still there 🤔

**Review by GeoffThorpeFT — APPROVED** [review]
Changes looks great to me but as you say there is something going on with the dubnium checks.

**Comment by nl-ria** [comment]
I have deleted the `.npmrc` file , but its still being used [here](https://github.com/Financial-Times/lambda-logger/blob/master/.circleci/config.yml#L103) - can someone explain its use to me in this context please? I didnt think it was required since we are using node v14 everywhere.

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,19 +4,17 @@ orbs:
   ft-snyk-orb: financial-times/ft-snyk-orb@0
 
 references:
-
-  container_config_carbon: &container_config_carbon
+  container_config_fermium: &container_config_fermium
     working_directory: ~/project/lambda-logger
     docker:
-      - image: circleci/node:8.10.0
+      - image: circleci/node:fermium # codename for nodejs active LTS v14
```
Comment (i-like-robots) [inline .circleci/config.yml]: I'd probably avoid using the codename as they're not widely known - if the comment is necessary then it's too complex 😄 

`node:14` should do the trick 👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -60,8 +57,8 @@ jobs:
       - *cache_node_modules
       - *persist_workspace
 
-  test_carbon:
-    <<: *container_config_carbon
+  test_fermium:
+    <<: *container_config_fermium
```
Comment (i-like-robots) [inline .circleci/config.yml]: Maybe replace `_carbon/_fermium` with just `_node12/_node14` for the same reason as above, not sure folks pay attention to release names.

File: .circleci/config.yml [hunk 3]
```diff
@@ -4,19 +4,17 @@ orbs:
...
 references:
-
-  container_config_carbon: &container_config_carbon
+  container_config_fermium: &container_config_fermium
     working_directory: ~/project/lambda-logger
     docker:
-      - image: circleci/node:8.10.0
+      - image: circleci/node:fermium # codename for nodejs active LTS v14
 
   container_config_dubnium: &container_config_dubnium
```
Comment (i-like-robots) [inline .circleci/config.yml]: We should probably swap `_dubnium`  with `_node12` as Node 10 is end-of-life.

File: .circleci/config.yml (hunk 2 above)
Comment (nl-ria) [reply .circleci/config.yml]: good point and I concur !

**Comment by nl-ria** [comment]
> I have deleted the `.npmrc` file , but its still being used [here](https://github.com/Financial-Times/lambda-logger/blob/master/.circleci/config.yml#L103) - can someone explain its use to me in this context please? I didnt think it was required since we are using node v14 everywhere.

Thoughts on this @i-like-robots ?

**Review by i-like-robots — APPROVED** [review]

**Pull request merged**